PyRoute/Calculation/RouteCalculation.py

```diff
@@ -166,5 +173,25 @@ def calculate_components(self):
             counter += 1
             self.components[counter] = len(component)
             for star in component:
-                star.component = counter
+                self.galaxy.star_mapping[star].component = counter
         return
+
+    def get_landmarks(self, index=False):
+        result = dict()
+
+        # Dig out landmarks for each connected component
+        # First landmark is the star with the biggest WTN in the component.
+        # Later landmark(s), if any, will probably be the star in the component furthest away from the closest existing
+        # landmark in that component.  If all the stars in a component are _already_ landmarks, return the previous
+        # landmark choice.
+        for component_id in self.components:
+            stars = [self.galaxy.star_mapping[item] for item in self.galaxy.stars]
+            stars = [item for item in stars if component_id == item.component]
```

Comment:
```suggestion
            stars = [item for item in self.galaxy.star_mapping.values() if component_id == item.component]
```
Again, no need to generate the stars list before iterating over it once, and then throwing it away.